plugins/default/src/Ide/Plugin/Eval/CodeLens.hs

```diff
@@ -0,0 +1,719 @@
+  -- let nfp = toNormalizedFilePath' fp
+  -- session :: HscEnvEq <-
+  --   liftIO $
+  --     runAction "runEvalCmd.ghcSession" state $
+  --       use_ GhcSession nfp
+  -- Right (ppContent, dflags) <- liftIO $ evalGhcEnv (hscEnv session) $ runExceptT $ preprocessor fp (Just $ textToStringBuffer content)
+  -- let text = decodeUtf8 $ stringBufferToByteString ppContent
+  -- liftIO $ dbg "PREPROCESSED CONTENT" text
+
+  -- Parse test sections
+  -- CHECK: add a Shake Action to return the result of parsing tests out of the current file?
+  let Right (setups, nonSetups) =
+        (splitSections . filter hasTests <$>)
+          . allSections
+          . tokensFrom
+          . T.unpack
+          $ text -- fp' (T.unpack text)
+  let tests = testsBySection nonSetups
+
+  cmd <- liftIO $ mkLspCommand plId evalCommandName "Evaluate=..." (Just [])
+  let lenses =
+        [ CodeLens testRange (Just cmd') Nothing
+          | (section, test) <- tests,
+            let (testRange, resultRange) = testRanges test
+                args = EvalParams (setups ++ [section]) _textDocument
+                cmd' =
+                  (cmd :: Command)
+                    { _arguments = Just (List [toJSON args]),
+                      _title =
+                        if trivial resultRange
+                          then "Evaluate..."
+                          else "Refresh..."
+                    }
+        ]
+
+  dbg "SETUPS" setups
+
+  dbg_ $
+    unwords
+      [ show (length tests),
+        "tests in",
+        show (length nonSetups),
+        "sections",
+        show (length setups),
+        "setups",
+        show (length lenses),
+        "lenses."
+      ]
+
+  now' <- liftIO getCurrentTime
+  dbg "TIME" (diffUTCTime now' now)
+
+  -- dbg "LENSES" lenses
+  return $ List lenses
+  where
+    trivial (Range p p') = p == p'
+
+testsBySection :: [Section] -> [(Section, Loc Test)]
+testsBySection sections =
+  [(section, test) | section <- sections, test <- sectionTests section]
+
+evalCommandName :: CommandId
+evalCommandName = "evalCommand"
+
+evalCommand :: PluginCommand
+evalCommand = PluginCommand evalCommandName "evaluate" runEvalCmd
+
+data EvalParams = EvalParams
+  { sections :: [Section],
+    module_  :: !TextDocumentIdentifier
+  }
+  deriving (Eq, Show, Generic, FromJSON, ToJSON)
+
+-- TODO: report errors that take place during
+runEvalCmd :: CommandFunction EvalParams
+runEvalCmd lsp state EvalParams {..} = withIndefiniteProgress lsp "Evaluating" Cancellable $ response' $ do
+  now <- liftIO getCurrentTime
+
+  let TextDocumentIdentifier {_uri} = module_
+  fp <- handleMaybe "uri" $ uriToFilePath' _uri
+  let nfp = toNormalizedFilePath' fp
+  contents <- liftIO $ getVirtualFileFunc lsp $ toNormalizedUri _uri
+  text <- handleMaybe "contents" $ virtualFileText <$> contents
+
+  session :: HscEnvEq <-
+    liftIO $
+      runAction "runEvalCmd.ghcSession" state $
+        use_ GhcSession nfp
```

Comment:
Why did you change this? It should be `GhcSessionDeps`

Reply:
Fixed.

Reply:
Could you make sure that there are no other merge accidents?

Reply:
Did you have a chance to check if there were any other merge accidents?

Reply:
By merge accidents you mean some other accidental change/regression with respect to your original code? 

I don't think so.